### ISSUE TITLE
Builds with ONEDNN_ENABLE_MAX_CPU_ISA=OFF crash

### ISSUE THREAD

**Issue by AngryLoki** [oneapi-src/oneDNN#1791]
# Summary
oneDNN was built like this:
```
	cmake -H. -Bbuild -GNinja \
	-DCMAKE_BUILD_TYPE=Debug \
	-DONEDNN_LIBRARY_TYPE=STATIC \
	-DONEDNN_BUILD_EXAMPLES=OFF \
	-DONEDNN_BUILD_TESTS=OFF \
	-DONEDNN_ENABLE_PRIMITIVE=MATMUL \
	-DONEDNN_BUILD_GRAPH=OFF \
	-DONEDNN_ENABLE_WORKLOAD=INFERENCE \
	-DONEDNN_ENABLE_PRIMITIVE_CPU_ISA=AVX512 \
	-DONEDNN_ENABLE_GEMM_KERNELS_ISA=NONE \
	-DONEDNN_ENABLE_PRIMITIVE_GPU_ISA= \
	-DONEDNN_VERBOSE=ON \
	-DONEDNN_ENABLE_JIT_PROFILING=OFF \
	-DONEDNN_ENABLE_ITT_TASKS=OFF \
	-DONEDNN_ENABLE_PRIMITIVE_CACHE=OFF \
	-DONEDNN_ENABLE_CPU_ISA_HINTS=OFF \
	-DONEDNN_ENABLE_MAX_CPU_ISA=OFF 
```

Attempt to build and run https://oneapi-src.github.io/oneDNN/page_cpu_sgemm_and_matmul_cpp.html fails with segfault. When oneDNN is built with `-DONEDNN_ENABLE_MAX_CPU_ISA=ON`, crash does not occur.

Crash happens in jit code, so I was unable to generate any meaningful stack trace...

# Version
3.3.3 / 3.3.4 / f11616d

**Comment by dzarukin** [comment]
Hi @AngryLoki, thank you for the report. My initial analysis tells that the feature itself is not working as intended which causes a crash in jit_reorder implementation. The plan is to update the documentation of this (and all similar) build options to reflect the intention, align the behavior with the documentation and improve the validation.

**Comment by dzarukin** [comment]
Must be fixed by [37a9f829](https://github.com/oneapi-src/oneDNN/commit/37a9f829b23f58293a4bf7b8c4e4cb5c2755ac1b)
